ouroboros-consensus/src/Ouroboros/Storage/VolatileDB/Impl.hs

```diff
@@ -279,16 +279,17 @@ reOpenDBImpl VolatileDBEnv{..} =
 getBlockImpl :: (IOLike m, Ord blockId)
              => VolatileDBEnv m blockId
              -> blockId
-             -> m (Maybe ByteString)
-getBlockImpl env@VolatileDBEnv{..} slot =
+             -> m (Maybe (SlotNo, ByteString))
+getBlockImpl env@VolatileDBEnv{..} blockId =
     modifyState env $ \hasFS@HasFS{..} st@InternalState{..} ->
-      case Map.lookup slot _currentRevMap of
+      case Map.lookup blockId _currentRevMap of
         Nothing -> return (st, Nothing)
-        Just InternalBlockInfo {..} ->  do
-          bs <- withFile hasFS ibFile ReadMode $ \hndl -> do
-                  _ <- hSeek hndl AbsoluteSeek (fromIntegral ibSlotOffset)
-                  hGetExactly hasFS hndl $ unBlockSize ibBlockSize
-          return (st, Just bs)
+        Just InternalBlockInfo {..} -> do
+          let size   = unBlockSize ibBlockSize
+              offset = AbsOffset ibSlotOffset
+          bs <- withFile hasFS ibFile ReadMode $ \hndl ->
+            hGetExactlyAt hasFS hndl size offset
```

Comment:
Nice, this is the new `pread` version right? (At least on Unix).

One of the next optimisations will want to be not opening these files each time. Though we'll have to think carefully about the number of open file handles, to keep that predictable so we can operate correctly near our file handle limit. Another nice one for the FS sim tests, we can see how many files we have open and check that's what we expected.

Reply:
Indeed. See #1278.